### ISSUE TITLE
IDE0044 False Negative In Partial Class

### ISSUE THREAD

**Issue by nnpcYvIVl** [dotnet/roslyn#29320]
**Version Used**: 
VS 15.8 Enterprise

**Steps to Reproduce**:
Load this code in the IDE
```csharp
namespace IDE0044_False_Negative
{
    public class Class
    {
        string foo;
    }

    public partial class PartialClass
    {
        string bar;
    }

    public partial class PartialClass { }
}
```

**Expected Behavior**:
foo and bar should both be flagged for IDE0044: Make field readonly

**Actual Behavior**:
bar is not flagged for IDE0044

**Comment by kuskmen** [comment]
Hey guys, 

I want to work on this one as my first contribution to the project. I think I am making some progress in the debugging so if there is anything more to be said, like some edge cases that probably has to be considered/tested that are not mentioned in the bug, please do share!

Regards,
kuskmen

**Comment by CyrusNajmabadi** [comment]
@kuskmen I believe @mavasani may have just rewritten this analyzer to no longer have the above problem.  You should check with him on this :)

**Comment by kuskmen** [comment]
@CyrusNajmabadi , thanks for the heads up, so I guess I will wait for him to reply in this thread? Unless you have something else in mind?

**Comment by CyrusNajmabadi** [comment]
That would probably be best.  though if you want to go exploring, feel free :)

**Comment by mavasani** [comment]
Yes, this has been fixed with https://github.com/dotnet/roslyn/pull/28918, which was merged last week. You will have to wait for Dev16 Preview1 to get the VS build with the fix though.

**Comment by kuskmen** [comment]
Very well then, I will pick something else.

**Comment by nnpcYvIVl** [comment]
@mavasani I am using VS 16.0.0 RC.2 and can still repro the issue. Can you confirm that it is fixed?

**Comment by mavasani** [comment]
@nnpcYvIVl I pasted the repro code in VS2019 Preview4 drop and do see the expected diagnostics. Kindly file a separate issue if you have a different repro with unexpected or missing diagnostics.

![image](https://user-images.githubusercontent.com/10605811/54366021-22090580-462d-11e9-8643-fd04f9639876.png)

**Comment by mavasani** [comment]
Closing this bug as the original repro is indeed working as expected in VS2019.

**Comment by nnpcYvIVl** [comment]
@mavasani Thanks! My mistake, it is working correctly now.
